remapping_commands.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+#requires vcf2bed, bowtie2, samtools, bedtools and reverse_strand.py (uses bamnostic)
+
+#This basic pipeline allows you to input two assemblies and a variant vcf file, 
+#and remap the variants to the new genome
+
+
+#example command:
+#time ./remapping_commands.sh -g droso_dm3.fasta -n GCA_000001215.4_Release_6_plus_ISO1_MT_genomic.fna -a GCA_000001215.4 -v droso_variants_renamed.vcf -o test.vcf
+
+
+#To be added:
+# -support for indels
+# -find a way to extract new assembly accession so it's not required as input? or just use new assembly file name?
+# -decide what to do in the case of new REF allele being the same as the variant allele (was A > G on old assembly, now is G > G on new assembly)
+
+#Completed:
+# -filter only SNPs (not indels) using bcftools filter
+# -copy filter etc columns from original vcf
+# -copy vcf header to the new vcf file, but regenerate contigs from the list of contigs in new assembly, and change reference tag
+# -take into account reverse strand
+# -create a command line version where you input file names
+# -% of remapped variants calculation
+
+
+oldgenome=''
+newgenome=''
+newgenomeaccession=''
+vcffile=''
+outfile=''
+
+while getopts 'g:n:a:v:o:' flag; do
+	case "${flag}" in
+		g) oldgenome="${OPTARG}";;
+		n) newgenome="${OPTARG}" ;;
+		a) newgenomeaccession="${OPTARG}" ;;
+		v) vcffile="${OPTARG}" ;;
+		o) outfile="${OPTARG}" ;;
+	esac
+done
+
+
+#1) Flanking sequence generation
+echo "-----------------------1) Flanking sequence generation-----------------------"
+#generate chrom sizes:
+samtools faidx $oldgenome 
+cut -f1,2 "$oldgenome".fai > "$oldgenome".chrom.sizes
+
+#filter SNPs only
+bcftools filter -i 'TYPE="snp"' $vcffile > snps_only.vcf
+
+#store header
+awk '$1 ~ /^#/ {print $0}' snps_only.vcf > vcf_header.txt
+
+#convert vcf to bed:
+vcf2bed < snps_only.vcf > variants.bed
+#the actual position of the variant is the second coordinate
+rm snps_only.vcf
+
+#generate the flanking sequence intervals
+bedtools flank -i variants.bed -g "$oldgenome".chrom.sizes -b 50 > flanking.bed
+rm variants.bed "$oldgenome".chrom.sizes
+
+#this outputs a flanking.bed file: for 1 variant we have the following lines:
+#[chr] [-50 pos] [variant pos-1] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+#[chr] [variant pos] [+50 pos] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+
+#we want 1 line per variant, with 1 interval to create the read:
+#[chr] [-50 pos] [+50 pos] [rsID] [qual] [REF allele] [ALT allele] [filter] [info]
+
+#fix rows: put both flanking sequences on 1 row so each read is 101 bases: [50 bases]-[variant]-[50 bases]
+awk 'NR == 1 {prev = $2} NR>1 && NR%2==0 {print $1"\t"prev"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7"\t"$8"\t"$9} NR%2 == 1 { prev = $2}' flanking.bed > flanking.corrected.bed
+rm flanking.bed
```

Comment:
You can use `bedtools slop` instead of `bedtools flank` to generate a two-way flanking sequence in one go